frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1442,29 +1473,35 @@ function ReaderFooter:_updateFooterText(force_repaint, force_recompute)
         return
     end
     local text = self:genFooterText()
-    if text then
-        self.footer_text:setText(text)
-    end
```

Comment:
You seem to have duplicated that single one 5 times below in all if/else branches. Any reason for removing it from here?
No other solution to avoid duplicating it? May be have it simpler by just keeping it here, with before a `if not text then text = "" end` if we ever can get `text=nil`.
Same for the generic `setMaxWidth(math.floor(self._saved_screen_width - 2 * self.settings.progress_margin_width))`: you could do it here for all case, even if in one case, you do a duplicate call to that (it does not cost much).

Reply:
I didn't know that I can use `setText()` before `setMaxWidth()`. Before first commit I used `self.footer_text.max_width` and I had to duplicate that to proper work. Later I notice that better to use method `setMaxWidth()`.

Thanks for tip. Now should be much cleaner.

Reply:
IIRC, I had a vague reason for skipping the setText entirely if text is nil/empty, I'd like to keep that, if possible ;).

Reply:
May be in https://github.com/koreader/koreader/pull/4850/commits/4e8841654044f4f05e6b28cfd556e0e739b7cb7c from #4850.
Looks like it was just to prevent some issue with nil being returned when text is empty - which I guess might no more be needed since my recent changes to TextWidget. Also, since it, setText and setMaxWidth costs nothing until draw time or getSize() is called - and there are some shortcuts when text is empty (not rendering anything, but just getting the height from the font).
Also, (quick) seeing no logical change as most tests where `or not text or text == ""` and became `or text == ""`

Reply:
I'm wary of piling on "almost nothing"s on page turns, even if the situation is better than it used to be ;).

Plus, I've used this to debug what the status bar was actually doing on every page turn re: disabled stuff.

Reply:
But: yeah, at least for the other branches (i.e., nil -> ""), you're right, this is slightly simpler ;).

Reply:
Given all the (more expensive) things readerfooter already does on page turn, this (really) nothing won't hurt. Furthermore if we also now do setMaxWidth (any one of them will call _xtext:free(), which is the only "expensive" stuff that could be done), so setText() is really a no-op.
The setText() could be just wrapped in a `if text then` as before and it wouldn't hurt (as it did not hurt before), if you really want - but I think that's less readable - my first reflex when reading the PR was actually worrying : why don't we reset setText() to empty, just to have no risk later in the branches to have the previous text displayed or previous width re-used. One would have to really read the followup code to be sure there's no risk of that happening (I did not re-read it :)  but I expect you to do that to find a good reason of keeping your trickery :)

Reply:
You're completely right, I'm more concerned with the setMaWidth than the setText by now ;).

Reply:
(sorry, did not see your 3rd post just before mine.)

Reply:
It's been too long since I played with readerfooter for me to tell whether hiding the setMaxWidth behind a text = "" would be enough/right, though.

My understanding is that it's only needed for visible modes that don't already have a custom setMaxWidth?

Reply:
(To explain my reasoning and the previous comments: the goal being to skip useless string formatting/layout when the status bar is invisible).

Reply:
> My understanding is that it's only needed for visible modes that don't already have a custom setMaxWidth?

setText and setMaxWidth are also no-op when the text or max_width are the same as the existing ones.
So, when text = "" always, when no mode, we'll just have these no-ops executed on a TextWidget with some already empty text that won't ever be sized/rendered (because of is_empty shortcut and paintTo skip) even if it were to be rendered by the complex readerfooter code or errors within it :)

Reply:
Cool! Did not know that ;).

Then everything's purrfect ;).